src/Client/Update/ARCitectUpdate.fs

```diff
@@ -63,6 +80,19 @@ module ARCitect =
                     ARCitect.Interop.InteropTypes.ARCFile.Investigation, ArcInvestigation.toJsonString 0 inv
                 | ArcFiles.Template template ->
                     ARCitect.Interop.InteropTypes.ARCFile.Template, Template.toJsonString 0 template
+                | ArcFiles.DataMap (datamapParent, datamap) ->
+                    let json =
+                        match datamapParent with
+                        | None ->
+                            let parentDataMap = ArcAssay.create("default", datamap = datamap)
+                            ArcAssay.toJsonString 0 parentDataMap
+                        | Some parent when parent.Parent = DataMapParent.Assay ->
+                            let parentDataMap = ArcAssay.create(parent.ParentId, datamap = datamap)
```

Comment:
no. this is ARCitect logic. Send ARCitect the parentype, the parentid and the datamap json. ARCitect will insert at the correct position with this info